content/en/blog/2024/otel-arrow-production/index.md

```diff
@@ -0,0 +1,513 @@
+metrics signals in production settings too, where OTAP users can expect 50% to
+70% improvement relative to OTLP for similar pipeline configurations.
+
+With our previous experimental results now validated in production, the
+OpenTelemetry Protocol with Apache Arrow Exporter and Receiver are considered
+ready for general use.
+
+## Exporter and Receiver Components
+
+Apache Arrow is a great technology partner for OpenTelemetry, and these projects
+have many aspects in common. Like OpenTelemetry, Apache Arrow features a shared
+data specification, a transport protocol, a cross-language API specification,
+and community-built SDKs for numerous runtime environments.
+
+Apache Arrow is by now a standard in the data processing industry for
+column-oriented data, popular for both interoperability and performance reasons.
+Here is how we use this exciting technology to form a compression bridge between
+OpenTelemetry collectors.
+
+The compression bridge consists of two OpenTelemetry collectors labeled Exporter
+and Receiver, or they could equally be two pools of load-balanced collectors.
+
+As described in this
+["OTEP" design document](https://github.com/open-telemetry/oteps/blob/main/text/0156-columnar-encoding.md),
+the Exporter converts arbitrary OpenTelemetry data into an Arrow record batch.
+The Arrow record batch is a block of memory, with a standardized layout, making
+it possible to exchange data across address spaces and virtual process
+boundaries without copying data. This "zero copy" approach is how Arrow enables
+calling into foreign functions and processor-specific execution "kernels"
+compiled in any language capable of operating over Arrow record batches.
+
+Standard OpenTelemetry Protocol data is structured as a hierarchy of Resource,
+Scope, and individual telemetry items. Before encoding spans, logs, and metrics,
+the distinct resource and scope values are identified, deduplicated, and encoded
+as separate structures in the OTel-Arrow record batch. Here we make other
+optimizations, such as sorting items to improve locality in the data.
+
+The Apache Arrow project specifies a protocol for inter-process communication
+(IPC). [Arrow IPC](https://arrow.apache.org/docs/format/DissociatedIPC.html) is
+a "dissociated" protocol, not tied to any particular transport method for
+transmitting a sequence of Arrow record batches. Apache Arrow has an opinionated
+RPC framework of its own—Arrow Flight—however it was natural, for our purposes
+to leverage existing gRPC-configuration, services, authorization extensions, and
+instrumentation in the OpenTelemetry Collector by embedding Arrow IPC inside
+gRPC streams. This design lets us benefit from heavily-optimized Arrow IPC
+functions that compress and encode repeated record batches while offering
+drop-in compatibility with OTLP/gRPC service configurations.
+
+The OTel-Arrow Receiver applies logic to reverse the transformation done in the
```

Comment:
```suggestion
The OTel-Arrow Receiver applies logic to reverse the transformation done in the
```

Is it OTAP or OTel-ARrow, or are these 2 different things?

Reply:
Thank you. I share your confusion and have removed the acronym "OTAP" to try and alleviate the confusion. Just like OpenTelemetry (a project name) has a protocol abbreviated OTLP, we were trying to distinguish the project named OTel-Arrow from the protocol it embodies, which is an encoding of for OTel data in Arrow record batches. I think it's less confusing overall to refer to OTel-Arrow and let it be inferred whether this means the project or its protocol. I will leave "OTAP" out of this post.